version: "3.11"
services:
  db:
    image: postgres
    restart: on-failure
    env_file:
      - .env
    volumes:
        - pg_data:/var/lib/postgresql/data/
    ports:
      - "5432:5432"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}" ]

  redis:
    image: redis
    ports:
      - '6379:6379'

  app:
    build: .
    tty: true
    ports:
      - '8000:8000'
    environment:
      - REDIS_HOST=redis
    env_file:
      - .env
    command: sh -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - .:/app

  celery:
    build: .
    tty: true
    command: celery -A config worker -l INFO
    env_file:
      - .env
    depends_on:
      - redis
      - app

  celery_beat:
    build: .
    tty: true
    command: celery -A config beat -l INFO -S django
    env_file:
      - .env
    depends_on:
      - redis
      - app

volumes:
  pg_data:
